src/features/network/components/network-select.tsx

```diff
@@ -3,37 +3,43 @@ import { Label } from '@/features/common/components/label'
 import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/features/common/components/select'
 import { cn } from '@/features/common/utils'
 import { useCallback } from 'react'
-import { useLocation, useNavigate } from 'react-router-dom'
-import { Urls } from '@/routes/urls'
+import { useLocation, useNavigate, useParams } from 'react-router-dom'
 
-const settingsUrl = Urls.Settings.build({})
+type NetworkSelectProps = {
+  showLabel?: boolean
+}
 
-export function NetworkSelect() {
+export function NetworkSelect({ showLabel = true }: NetworkSelectProps) {
   const [selectedNetwork, setSelectedNetwork] = useSelectedNetwork()
   const networkConfigs = useNetworkConfigs()
+  const { networkId: currentNetworkId } = useParams()
   const navigate = useNavigate()
   const location = useLocation()
 
   const handleNetworkChange = useCallback(
-    async (value: string) => {
-      if (location.pathname !== settingsUrl) {
-        navigate(settingsUrl)
+    (newNetworkId: string) => {
+      const currentPath = location.pathname
+      setSelectedNetwork(newNetworkId)
+      if (currentNetworkId) {
+        const newUrl = currentPath.replace(currentNetworkId, newNetworkId)
+        navigate(newUrl)
       }
-      await setSelectedNetwork(value)
     },
-    [location.pathname, navigate, setSelectedNetwork]
+    [currentNetworkId, location.pathname, navigate, setSelectedNetwork]
   )
 
   return (
-    <div className={cn('flex w-48 flex-col')}>
-      <Label htmlFor="network" className={cn('ml-0.5 mb-2')}>
-        Active network
-      </Label>
+    <div className={cn('flex flex-col')}>
+      {showLabel && (
+        <Label htmlFor="network" className={cn('ml-0.5 mb-2')}>
+          Active network
+        </Label>
+      )}
       <Select onValueChange={handleNetworkChange} value={selectedNetwork}>
```

Comment:
The focus ring looks like it's getting cut off on the left side. See pic.

<img width="634" alt="Screenshot 2024-11-22 at 11 37 50 PM" src="https://github.com/user-attachments/assets/9b7b308e-a1fc-49ca-b223-74508c998277">

Reply:
fixed 
<img width="339" alt="image" src="https://github.com/user-attachments/assets/24a7baa7-0e53-4f09-bf4c-328854f8ff86">